Log4j 2.9.0 crashes at runtime

```
Expanding battlesnake.war into /home/jaxson/Documents/Development/src/Java/Battlesnake/battle-snake-2019/build/target/tomcat.8080/webapps/expanded
Adding Context  for /home/jaxson/Documents/Development/src/Java/Battlesnake/battle-snake-2019/build/target/tomcat.8080/webapps/expanded
Mar. 05, 2019 10:10:28 P.M. org.apache.coyote.AbstractProtocol init
INFO: Initializing ProtocolHandler ["http-nio-8080"]
Mar. 05, 2019 10:10:28 P.M. org.apache.tomcat.util.net.NioSelectorPool getSharedSelector
INFO: Using a shared selector for servlet write/read
Mar. 05, 2019 10:10:28 P.M. org.apache.catalina.core.StandardService startInternal
INFO: Starting service Tomcat
Mar. 05, 2019 10:10:28 P.M. org.apache.catalina.core.StandardEngine startInternal
INFO: Starting Servlet Engine: Apache Tomcat/8.5.11
Mar. 05, 2019 10:10:28 P.M. org.apache.catalina.startup.ContextConfig getDefaultWebXmlFragment
INFO: No global web.xml found
Mar. 05, 2019 10:10:29 P.M. org.apache.jasper.servlet.TldScanner scanJars
INFO: At least one JAR was scanned for TLDs yet contained no TLDs. Enable debug logging for this logger for a complete list of JARs that were scanned but no TLDs were found in them. Skipping unneeded JARs during scanning can improve startup time and JSP compilation time.
WARNING: sun.reflect.Reflection.getCallerClass is not supported. This will impact performance.
Mar. 05, 2019 10:10:29 P.M. org.apache.catalina.core.StandardContext filterStart
SEVERE: Exception starting filter Response Time
java.lang.ExceptionInInitializerError
        at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
        at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
        at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490)
        at java.base/java.lang.Class.newInstance(Class.java:584)
        at org.apache.catalina.core.DefaultInstanceManager.newInstance(DefaultInstanceManager.java:119)
        at org.apache.catalina.core.ApplicationFilterConfig.getFilter(ApplicationFilterConfig.java:258)
        at org.apache.catalina.core.ApplicationFilterConfig.<init>(ApplicationFilterConfig.java:105)
        at org.apache.catalina.core.StandardContext.filterStart(StandardContext.java:4572)
        at org.apache.catalina.core.StandardContext.startInternal(StandardContext.java:5215)
        at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:150)
        at org.apache.catalina.core.ContainerBase$StartChild.call(ContainerBase.java:1419)
        at org.apache.catalina.core.ContainerBase$StartChild.call(ContainerBase.java:1409)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.lang.UnsupportedOperationException: No class provided, and an appropriate one cannot be found.
        at org.apache.logging.log4j.LogManager.callerClass(LogManager.java:525)
        at org.apache.logging.log4j.LogManager.getLogger(LogManager.java:550)
        at org.apache.logging.log4j.LogManager.getLogger(LogManager.java:537)
        at com.battlesnake.filters.ResponseTimeFilter.<clinit>(ResponseTimeFilter.java:24)
        ... 17 more

Mar. 05, 2019 10:10:29 P.M. org.apache.catalina.core.StandardContext startInternal
SEVERE: One or more Filters failed to start. Full details will be found in the appropriate container log file
Mar. 05, 2019 10:10:29 P.M. org.apache.catalina.core.StandardContext startInternal
SEVERE: Context [] startup failed due to previous errors
SEVERE: Context [] failed in [org.apache.catalina.core.StandardContext] lifecycle. Allowing Tomcat to shutdown.
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by org.apache.catalina.loader.WebappClassLoaderBase (file:/home/jaxson/Documents/Development/src/Java/Battlesnake/battle-snake-2019/build/server/webapp-runner-8.5.11.3.jar) to field java.lang.Thread.threadLocals
WARNING: Please consider reporting this to the maintainers of org.apache.catalina.loader.WebappClassLoaderBase
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
Mar. 05, 2019 10:10:29 P.M. org.apache.coyote.AbstractProtocol start
INFO: Starting ProtocolHandler [http-nio-8080]

```

--- a/build.gradle
+++ b/build.gradle
@@ -33,10 +33,10 @@ war.mustRunAfter clean
 ext {
     gsonVersion = '2.8.5'
     junitVersion = '5.4.0'
+    log4jVersion = '2.11.2'
     tomcatVersion = '9.0.16'
     webappRunnerVersion = "$tomcatVersion.0"
     wsVersion = '2.1'
-    log4jVersion = '2.8.2'
     webappRunnerVersion = '8.5.34.0'
 }
 

--- a/build.gradle
+++ b/build.gradle
@@ -33,7 +33,8 @@ war.mustRunAfter clean
 ext {
     gsonVersion = '2.8.5'
     junitVersion = '5.4.0'
-    servletVersion = '4.0.1'
+    tomcatVersion = '9.0.16'
+    webappRunnerVersion = "$tomcatVersion.0"
     wsVersion = '2.1'
     log4jVersion = '2.8.2'
     webappRunnerVersion = '8.5.34.0'
@@ -49,8 +50,8 @@ repositories {
 dependencies {
     compile "com.google.code.gson:gson:$gsonVersion"
     compile "com.google.code.gson:gson-extras:$gsonVersion"
-    compile "javax.servlet:javax.servlet-api:$servletVersion"
-    compile "com.github.jsimone:webapp-runner:$webappRunnerVersion"
+    compile "org.apache.tomcat:tomcat-servlet-api:$tomcatVersion"
+    providedCompile "com.github.jsimone:webapp-runner:$webappRunnerVersion"
     compile "javax.ws.rs:javax.ws.rs-api:$wsVersion"
     compile "org.apache.logging.log4j:log4j-api:$log4jVersion"
     compile "org.apache.logging.log4j:log4j-core:$log4jVersion"

